src/mito2/src/worker/handle_open.rs

```diff
@@ -39,21 +39,23 @@ impl<S: LogStore> RegionWorkerLoop<S> {
         if self.regions.is_region_exists(region_id) {
             return Ok(Output::AffectedRows(0));
         }
-
+        let object_store = if let Some(storage_name) = request.options.get("storage") {
+            self.object_store_manager
+                .find(storage_name)
+                .context(ObjectStoreNotFoundSnafu {
+                    object_store: storage_name.to_string(),
+                })?
+        } else {
+            self.object_store_manager.default_object_store()
+        };
         // Check if this region is pending drop. And clean the entire dir if so.
         if !self.dropping_regions.is_region_exists(region_id)
-            && self
-                .object_store_manager
-                .default_object_store()
+            && object_store
```

Comment:
Maybe we should use the `object_store` of the region in `dropping_regions`?
```rust
if let Some(region) = self.dropping_regions.get_region(region_id) {
    // use region's object store
}
```

Reply:
```Rust
        // Check if this region is pending drop. And clean the entire dir if so.
        if !self.dropping_regions.is_region_exists(region_id)
            && object_store
                .is_exist(&join_path(&request.region_dir, DROPPING_MARKER_FILE))
                .await
                .context(OpenDalSnafu)?
```

I guess that the above condition is true when both the following conditions is true
* the deleted region isn't yet added to the `dropping_regions`
and
*  the dir of the deleted region is marked as deleted

I guess this means that we need to get the `object_store` from the `object_store_manager`, not `dropping_regions` since the deleted region isn't yet added to this.